main/input_default.cpp

```diff
@@ -598,7 +598,8 @@ Input::CursorShape InputDefault::get_default_cursor_shape() {
 
 void InputDefault::set_default_cursor_shape(CursorShape p_shape) {
 	default_shape = p_shape;
-	OS::get_singleton()->set_cursor_shape((OS::CursorShape)p_shape);
+	// Triggers _gui_input_event, which then does the right thing.
+	warp_mouse_position(mouse_pos);
```

Comment:
This solution is not too elegant and I'm not sure if this will works for windows and mac, windows's doc don't say if `SetCursorPos` will trigger event and mac docs say that `CGWarpMouseCursorPosition` will not.
https://docs.microsoft.com/en-us/windows/desktop/api/winuser/nf-winuser-setcursorpos
https://developer.apple.com/documentation/coregraphics/1456387-cgwarpmousecursorposition?language=objc

Reply:
Probably I can test on windows but we will need someone to test on mac.

Reply:
Any better Idea how to get the event, or how to check we are not over Control node? Alternatively we could document that the shape change happens after a mouse motion, but that's also not nice.

Reply:
Well, to get this working for all platforms, an alternative could be call "send event" method in the last line of `set_cursor_shape` for all platforms to dispatch an mouse motion event.

[windows] https://docs.microsoft.com/en-us/windows/desktop/api/winuser/nf-winuser-sendinput
[x11] https://www.x.org/archive/X11R7.5/doc/man/man3/XSendEvent.3.html
[osx] https://developer.apple.com/documentation/appkit/nswindow/1419228-sendevent?language=objc

Reply:
> in the last line of set_cursor_shape

That won't work. The actual problem is that `XDefineCursor` immediately changes the cursor shape
https://www.x.org/releases/X11R7.5/doc/man/man3/XUndefineCursor.3.html

`set_cursor_shape` is already called in` _gui_input_event` exactly as we need it = only outside the Control nodes.

https://github.com/godotengine/godot/blob/af5c30acb2eee54ce78e3dc713eea97bd3ddc18d/scene/main/viewport.cpp#L1901-L1904

All we have to do is trigger the GUI event, otherwise nothing happens if you call `set_default_cursor_shape` with the cursor lying in the viewport - well, until you move the mouse.

Reply:
Use this then:
```
Ref<InputEventMouseMotion> mm;
mm.instance();
mm->set_position(mouse_pos);
parse_input_event(mm);
```
instead of `warp_mouse_position(mouse_pos);`, this should works for all platforms.

Reply:
Excellent, tested on Linux, cross-compiled Windows and 3D  (^_^).

![peek6](https://user-images.githubusercontent.com/4047289/45589706-87a06180-b92a-11e8-8941-b403472d4e7a.gif)

Reply:
:+1: